### PR TITLE
fix: wait for proxy to start

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/wda/iproxy.js [hunk 1]
```diff
@@ -30,7 +31,12 @@ class iProxy {
         connection.destroy();
       }
     });
+    const status = new B((resolve, reject) => {
```
Comment (mykola-mokhnach) [inline lib/wda/iproxy.js]: should we also reject this promise if an error happens in one of the above callbacks for connection and socket?

File: lib/wda/iproxy.js [hunk 2]
```diff
@@ -30,7 +31,12 @@ class iProxy {
...
+      this.serverSocket.once('listening', resolve);
```
Comment (mykola-mokhnach) [inline lib/wda/iproxy.js]: `once` can be replaced with `on`. The actual promise implementation will ignore all the further resolve/reject calls if it was called once

Comment (umutuzgur) [reply lib/wda/iproxy.js]: I know but `once` seemed more appropriate here since it will be called only once

File: lib/wda/iproxy.js (hunk 1 above)
Comment (umutuzgur) [reply lib/wda/iproxy.js]: That shouldn't be necessary. Those callbacks are used when the port is available

**Pull request merged**